ingest/sqs/system/delivery/http/system_http_handler.go

```diff
@@ -1,17 +1,130 @@
 package http
 
 import (
+	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
 	_ "net/http/pprof"
+	"os"
+
+	"go.uber.org/zap"
+
+	"github.com/go-redis/redis"
+
+	"github.com/osmosis-labs/osmosis/v20/ingest/sqs/domain/mvc"
+	"github.com/osmosis-labs/osmosis/v20/ingest/sqs/log"
 
 	"github.com/labstack/echo"
 )
 
+type SystemHandler struct {
+	logger    log.Logger
+	host      string // Host for the GRPC gateway
+	grpcPort  string // GRPC gateway port
+	redisHost string // Redis host
+	redisPort string // Redis port
+	CIUsecase mvc.ChainInfoUsecase
+}
+
 // NewSystemHandler will initialize the /debug/ppof resources endpoint
-func NewSystemHandler(e *echo.Echo) {
+func NewSystemHandler(e *echo.Echo, logger log.Logger, us mvc.ChainInfoUsecase) {
+	// GRPC Gateway configuration
+	host := getEnvOrDefault("GRPC_GATEWAY_HOST", "localhost")
+	grpcPort := getEnvOrDefault("GRPC_GATEWAY_PORT", "1317")
+
+	// Redis configuration
+	redisHost := getEnvOrDefault("REDIS_HOST", "localhost")
+	redisPort := getEnvOrDefault("REDIS_PORT", "6379")
+
+	handler := &SystemHandler{
+		logger:    logger,
+		host:      host,
+		grpcPort:  grpcPort,
+		redisHost: redisHost,
+		redisPort: redisPort,
+		CIUsecase: us,
+	}
 
 	e.GET("/debug/pprof/*", echo.WrapHandler(http.DefaultServeMux))
+	e.GET("/health", handler.GetHealthStatus)
 
 	// // Register pprof handlers on "/debug/pprof"
 	// e.GET("/debug/pprof/*", echo.WrapHandler(http.HandlerFunc(pprof.Index)))
 }
+
+// GetHealthStatus handles health check requests for both GRPC gateway and Redis
+func (h *SystemHandler) GetHealthStatus(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	// Check GRPC Gateway status
+	grpcStatus := "running"
+	url := fmt.Sprintf("http://%s:%s/status", h.host, h.grpcPort)
+	resp, err := http.Get(url)
+	if err != nil {
+		grpcStatus = "down"
+		h.logger.Error("Error checking GRPC gateway status", zap.Error(err))
+	} else {
+		defer resp.Body.Close()
+	}
+
+	// Check the latest height from chain info use case
+	latestHeight, err := h.CIUsecase.GetLatestHeight(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Parse the response from the GRPC Gateway status endpoint
+	var statusResponse struct {
+		Result struct {
+			SyncInfo struct {
+				LatestBlockHeight string `json:"latest_block_height"`
+			} `json:"sync_info"`
+		} `json:"result"`
```

Comment:
Is there some tendermint struct we can import instead of hardcoding these?

Reply:
I tried changing this to import and use tendermint directly, the cometbft version we have imported and the actual tendermint version we are running on seems to have diference ( maybe due to diff between comet bft version between main and mainnet), retained this for now so that we can further test this

Reply:
OK